### PR TITLE
docs: ZkProgram docs and code, remove experimental, add name argument

### PR DESCRIPTION
**Pull request by barriebyron**
This PR addresses https://github.com/o1-labs/docs2/issues/711 

update: thanks @garwalsh for guidance and @iregina for adding the `name` argument to the code snippets

### CODE REVIEW

**Comment by barriebyron** [comment]
another idea for ZkProgram docs https://app.zenhub.com/workspaces/zkapps-product-eng-6130fedb3b0fc600123d8796/issues/gh/o1-labs/docs2/548

**Comment by iregina** [comment]
> another idea for ZkProgram docs https://app.zenhub.com/workspaces/zkapps-product-eng-6130fedb3b0fc600123d8796/issues/gh/o1-labs/docs2/548

Maybe this can be a separate pull request? Cause right now this one already has a lot!

**Comment by iregina** [comment]
Let's start with the first task - add a name argument to the code snippets: 

- [ ] /Users/regina/code/docs2/docs/zkapps/o1js/custom-tokens.mdx
- [ ] /Users/regina/code/docs2/docs/zkapps/o1js-reference/modules.md
- [ ] /Users/regina/code/docs2/docs/zkapps/o1js-reference/modules.md

**Comment by iregina** [comment]
@barriebyron I prefer making a separate pull request for updating the example zkapps - could we possibly merge these ones first?

File: docs/zkapps/o1js/recursion.mdx [hunk 1]
```diff
@@ -32,17 +32,9 @@ Kimchi, the proof system that backs o1js, supports arbitrary infinite recursive
 
 More generally, you can use recursion to verify any zero-knowledge program as part of your zkApp.
```
Comment (garwalsh) [inline docs/zkapps/o1js/recursion.mdx:34]: :::note
As of o1js v0.14.0, zkProgram has moved out of the Experimental namespace and is now available as a top-level import directly. 
Experimental.ZkProgram has been deprecated. 

If you are experiencing issues with zkProgram as documented below, please update o1js to the latest version
:::

**Review by garwalsh — APPROVED** [review]

**Pull request merged**